src/java/org/apache/cassandra/schema/SchemaKeyspace.java

```diff
@@ -58,8 +58,10 @@
 
 /**
  * system_schema.* tables and methods for manipulating them.
+ * 
+ * Please notice this class is _not_ thread safe. It should be accessed through {@link org.apache.cassandra.schema.Schema}. See CASSANDRA-16856/16996
  */
-public final class SchemaKeyspace
+final class SchemaKeyspace
```

Comment:
nit: might be a good place for a `@NotThreadSafe`?

Reply:
+100000